gravitee-am-management-api/gravitee-am-management-api-rest/src/test/java/io/gravitee/am/management/handlers/management/api/resources/CertificatesResourceTest.java

```diff
@@ -68,7 +70,64 @@ public void shouldGetCertificates() {
         assertEquals(HttpStatusCode.OK_200, response.getStatus());
 
         final List<Certificate> responseEntity = readEntity(response, List.class);
-        assertTrue(responseEntity.size() == 2);
+        assertEquals(2, responseEntity.size());
+    }
+
+    @Test
+    public void shouldGetCertificates_with_usages() {
+        final String domainId = "domain-1";
+        final Domain mockDomain = new Domain();
+        mockDomain.setId(domainId);
+
+        final Certificate mockCertificate = new Certificate();
+        mockCertificate.setId("certificate-1-id");
+        mockCertificate.setName("certificate-1-name");
+        mockCertificate.setDomain(domainId);
+        mockCertificate.setConfiguration("{\"jks\":\"keystore.jks\",\"storepass\":\"password\",\"alias\":\"abc\",\"keypass\":\"password\",\"use\":[\"sig\", \"enc\"]}");
+
+        final Certificate mockCertificate2 = new Certificate();
+        mockCertificate2.setId("certificate-2-id");
+        mockCertificate2.setName("certificate-2-name");
+        mockCertificate2.setDomain(domainId);
+        mockCertificate2.setConfiguration("{\"jks\":\"keystore.jks\",\"storepass\":\"password\",\"alias\":\"abc\",\"keypass\":\"password\",\"use\":[\"mtls\"]}");
+
+        final Certificate mockCertificate3 = new Certificate();
+        mockCertificate3.setId("certificate-3-id");
+        mockCertificate3.setName("certificate-3-name");
+        mockCertificate3.setDomain(domainId);
+        mockCertificate3.setConfiguration("{\"jks\":\"keystore.jks\",\"storepass\":\"password\",\"alias\":\"abc\",\"keypass\":\"password\"}");
+
+        final Certificate mockCertificate4 = new Certificate();
+        mockCertificate4.setId("certificate-4-id");
+        mockCertificate4.setName("certificate-4-name");
+        mockCertificate4.setDomain(domainId);
+        mockCertificate4.setConfiguration(null);
+
+
+        doReturn(Maybe.just(mockDomain)).when(domainService).findById(domainId);
+        doReturn(Flowable.just(mockCertificate, mockCertificate2, mockCertificate3, mockCertificate4)).when(certificateService).findByDomain(domainId);
+        doReturn(Flowable.empty()).when(applicationService).findByCertificate(anyString());
+
+        final Response response = target("domains").path(domainId).path("certificates").request().get();
+        assertEquals(HttpStatusCode.OK_200, response.getStatus());
+
+        final List<LinkedHashMap> responseEntity = readEntity(response, List.class);
```

Comment:
There is readListEntity method. You can use it like this:  readListEntity(reponse, CertificateEntity.class)